Packs/Cyberint/Integrations/Cyberint/Cyberint.py

```diff
@@ -313,30 +471,58 @@ def fetch_incidents(client: Client, last_run: Dict[str, int],
     #  Send the API request to fetch the alerts.
     alerts = client.list_alerts('1', max_fetch, datetime.strftime(last_fetch, DATE_FORMAT),
                                 datetime.strftime(datetime.now(), DATE_FORMAT), None, None,
                                 fetch_environment, fetch_status, fetch_severity, fetch_type)
-    for alert in alerts.get('alerts', []):
+
+    for alert_object in alerts.get('alerts', []):
+        alert = dict(alert_object)
         #  Create the XS0AR incident.
-        alert_created_time = datetime.strptime(alert.get('created_date'), '%Y-%m-%dT%H:%M:%S')
+        alert_created_time = datetime.strptime(alert.get('created_date'), '%Y-%m-%dT%H:%M:%S')  # type: ignore
         alert_id = alert.get('ref_id')
         alert_title = alert.get('title')
-        alert_csv_id = alert.get('alert_data', {}).get('csv', {}).get('id', '')
-        if alert_csv_id:
-            extracted_csv_data = extract_data_from_csv_stream(client, alert_id,
-                                                              alert_csv_id)
-            alert['alert_data']['csv'] = extracted_csv_data
+        attachments = []
+        incident_attachments = []
+
+        attachments_keys = {'attachment': [["attachments"], ["alert_data", "screenshot"], ["alert_data", "csv"]],
+                            'analysis_report': [['analysis_report']]}
+        for attachment_type, attachments_path in attachments_keys.items():
+            for path in attachments_path:
+                current_attachments = dict_safe_get(alert, path, default_return_value=[])
+                attachment_list = current_attachments if isinstance(current_attachments, list) else [
+                    current_attachments]
+                # Retrieve alert Incident attachments files - Attachments, CSV, Screenshot, and Analysis report.
+                current_incident_attachments = get_alert_attachments(client,
+                                                                     attachment_list,
+                                                                     attachment_type,
+                                                                     alert_id)  # type: ignore
+
+                incident_attachments.extend(current_incident_attachments)
+                for tmp_attachment in attachment_list:
+                    if tmp_attachment:
+                        attachments.append(tmp_attachment)
+
+        alert["attachments"] = attachments
+
         incident = {
             'name': f'Cyberint alert {alert_id}: {alert_title}',
             'occurred': datetime.strftime(alert_created_time, DATE_FORMAT),
             'rawJSON': json.dumps(alert),
-            'severity': SEVERITIES.get(alert.get('severity', 'low'), 1)
+            'severity': SEVERITIES.get(alert.get('severity', 'low'), 1),
+            'attachment': incident_attachments
         }
         incidents.append(incident)
+
+        alert_csv_id = alert.get('alert_data', {}).get('csv', {}).get('id', '')

```

Comment:
You're right, this line should be before the creation of the incident.
I added the creation of the incident attachment files before this command because it changes some data I need in the alert.